### PR TITLE
ESIESI-11839 Gohan sometimes crashes when starting parallel tests

### PR DESCRIPTION
**Pull request by mmatczuk**
* extension manager map protected
* package for handling singletons based on work on schema.Manager

### CODE REVIEW

**Comment by mmatczuk** [comment]
@marcin-ptaszynski @Zimnx please review

File: singleton/singleton.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2015 NTT Innovation Institute, Inc.
```
Comment (p-kozlowski) [inline singleton/singleton.go]: 2017

File: extension/extension.go [hunk 2]
```diff
@@ -66,17 +76,16 @@ func (manager *Manager) GetEnvironment(schemaID string) (env Environment, ok boo
 
 //GetManager gets manager
 func GetManager() *Manager {
-	if manager == nil {
-		manager = &Manager{
+	return singleton.Get("extension/manager", func() interface{} {
```
Comment (p-kozlowski) [inline extension/extension.go]: make the string a const

File: schema/manager.go [hunk 3]
```diff
@@ -507,84 +507,31 @@ func (manager *Manager) ClearExtensions() {
...
-		if manager == nil {
-			manager = newManager()
-		}
-		return manager
-	case ScopeGLSSingleton:
-		m := gls.Get("manager")
-		if m == nil {
-			m = newManager()
-			gls.Set("manager", m)
+	return singleton.Get("schema/manager", func() interface{} {
```
Comment (p-kozlowski) [inline schema/manager.go]: const

**Pull request merged**